src/components/settings/SettingsForm.tsx

```diff
@@ -0,0 +1,316 @@
+                  <EyeOffIcon className="size-4 shrink-0 text-gray-400 group-hover:text-white" />
+                )}
+              </button>
+            </div>
+          </div>
+
+          <div className="col-span-full">
+            <div>
+              <Label label="PIN (recommended)" htmlFor="pin" />
+              <Description>
+                Setup your four digit PIN to encrypt your API key stored on your
+                device. You will be asked to enter this PIN on first request
+                sent to Plainly every time you restart the extension. If you
+                don&apos;t want to use a PIN, you can leave this field empty.
+              </Description>
+            </div>
+            <div className="mt-2 grid grid-cols-6 items-center justify-items-center gap-y-2">
+              <PinInput
+                className="col-start-3 col-end-5 row-start-1"
+                pin={pin}
+                showPin={showPin}
+                onChange={setPin}
+              />
+              <div className="col-start-3 col-end-5 row-start-2 flex items-center justify-center gap-x-1">
+                <div className="bg-white/10 h-[1px] w-8" />
+                <button
+                  type="button"
+                  onClick={() => setShowPin((prev) => !prev)}
+                  className="cursor-pointer w-4 h-4 flex items-center justify-center group mx-3"
+                >
+                  {showPin ? (
+                    <EyeIcon className="size-4 shrink-0 text-gray-400 group-hover:text-white" />
+                  ) : (
+                    <EyeOffIcon className="size-4 shrink-0 text-gray-400 group-hover:text-white" />
+                  )}
+                </button>
+                <div className="bg-white/10 h-[1px] w-8" />
+              </div>
+              <PinInput
+                className="col-start-3 col-end-5 row-start-3"
+                pin={confirmPin}
+                showPin={showPin}
+                onChange={setConfirmPin}
+              />
+            </div>
+          </div>
+        </div>
+      </div>
+
+      <Button
+        className="float-right"
+        disabled={loading || !apiKey}
+        loading={loading}
+      >
+        Save
+      </Button>
+
+      {notification && (
+        <Notification
+          title={notification.title}
+          type={notification.type}
+          description={notification.description}
+          onClose={clearNotification}
+        />
+      )}
+    </form>
+  );
+}
+
+export function PinInput({
+  pin,
+  onChange,
+  showPin,
+  className,
+}: {
+  pin: Pin | undefined;
+  onChange: (pin: Pin) => void;
+  showPin?: boolean;
+  className?: string;
+}) {
+  // Create refs for each input
+  const firstInputRef = useRef<HTMLInputElement>(null);
+  const secondInputRef = useRef<HTMLInputElement>(null);
+  const thirdInputRef = useRef<HTMLInputElement>(null);
+  const fourthInputRef = useRef<HTMLInputElement>(null);
+
+  const changeDigit = useCallback(
```

Comment:
seems not enough, if there is any input I don't have all text selected:

![image](https://github.com/user-attachments/assets/c358ffa4-2e1b-4cc3-92db-530ff62d447e)